e2e/test_optimize_build.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+# -*- indent-tabs-mode: nil; tab-width: 2; sh-indentation: 2; -*-
+
+set -x
+set -e
+set -o pipefail
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+OUTDIR="$(dirname "$SCRIPTDIR")/e2e-output"
+
+rm -rf "$OUTDIR"
+mkdir "$OUTDIR"
+
+tox -e e2e -n -r
+source .tox/e2e/bin/activate
+
+fromager \
+  --sdists-repo="$OUTDIR/sdists-repo" \
+  --wheels-repo="$OUTDIR/wheels-repo" \
+  --work-dir="$OUTDIR/work-dir" \
+  --log-file="$OUTDIR/logfile.txt" \
+  bootstrap 'stevedore==5.2.0'
+
+pass=true
+
+if ! grep -q "already resolved sdist version not found" "$OUTDIR/logfile.txt"; then
```

Comment:
When you remove the log line that emits this message, this part of the test would have to change. You could look for the same "has sdist version" message that is checked on line 41, but expect to not find it (and so fail if you do).